pkg/nificlient/client.go

```diff
@@ -349,7 +350,7 @@ func (n *nifiClient) nodeDtoByNodeId(nId int32) *nigoapi.NodeDto {
 	for id := range n.nodes {
 		nodeDto := n.nodes[id]
 		// Check if the Cluster Node uri match with the one associated to the NifiCluster nodeId searched
-		if fmt.Sprintf("%s:%d", nodeDto.Address, nodeDto.ApiPort) == fmt.Sprintf(n.opts.NodeURITemplate, nId) {
+		if strings.Split(nodeDto.Address, ".")[0] == strings.Split(fmt.Sprintf(n.opts.NodeURITemplate, nId), ".")[0] {
```

Comment:
Have you confirmed that this change works for both headless _and_ non-headless configurations?

Reply:
It does, though I was not able to give this PR enough time to feel comfortable stewarding it. I will leave it here as a draft in case someone else elects to pick up the mantle until I'm able to log a proper issue.